### PR TITLE
ggml: load all backends from a user-provided search path

### PR DESCRIPTION
**Pull request by giladgd**
This PR makes it possible to load all backends from a given search path instead of the default search paths.

### CODE REVIEW

File: ggml/include/ggml-backend.h [hunk 1]
```diff
@@ -228,6 +228,7 @@ extern "C" {
     GGML_API void               ggml_backend_unload(ggml_backend_reg_t reg);
     // Load all known backends from dynamic libraries
     GGML_API void               ggml_backend_load_all(void);
+    GGML_API void               ggml_backend_load_all_in_search_path(const char * search_path);
```
Comment (ngxson) [inline ggml/include/ggml-backend.h:231]: Just thinking, `ggml_backend_load_all_from_path` may be a better name.

Comment (giladgd) [reply ggml/include/ggml-backend.h]: Done

File: ggml/src/ggml-backend-reg.cpp [hunk 2]
```diff
@@ -509,21 +519,25 @@ static ggml_backend_reg_t ggml_backend_load_best(const char * name, bool silent)
 }
 
 void ggml_backend_load_all() {
+    ggml_backend_load_all_from_path(NULL);
```
Comment (slaren) [inline ggml/src/ggml-backend-reg.cpp:522]:
```suggestion
    ggml_backend_load_all_from_path(nullptr);
```

**Review by slaren — APPROVED** [review]

File: ggml/src/ggml-backend-reg.cpp (hunk 2 above)
Comment (giladgd) [reply ggml/src/ggml-backend-reg.cpp]: Done

**Pull request merged**